crypto/encoding/proto/types.proto

```diff
@@ -0,0 +1,45 @@
+syntax = "proto3";
+package tendermint.crypto.encoding.proto;
+
+option go_package = "github.com/tendermint/tendermint/crypto/encoding/proto";
+
+import "third_party/proto/gogoproto/gogo.proto";
+import "third_party/proto/cosmos-proto/cosmos.proto";
+
+option (gogoproto.populate_all) = true;
+option (gogoproto.equal_all)    = true;
+option (gogoproto.testgen_all)  = true;
+
+message PubKey {
+  option (cosmos_proto.interface_type) = "*github.com/tendermint/tendermint/crypto.PubKey";
+  oneof key {
+    bytes ed25519 = 1
+        [(gogoproto.casttype) = "github.com/tendermint/tendermint/crypto/ed25519.PubKey"];
+    bytes secp256k1 = 2
+        [(gogoproto.casttype) = "github.com/tendermint/tendermint/crypto/secp256k1.PubKey"];
+    bytes sr25519 = 3
+        [(gogoproto.casttype) = "github.com/tendermint/tendermint/crypto/sr25519.PubKey"];
+    // TODO: implement
+    // PubKeyMultiSigThreshold multisig = 4;
+    // [(gogoproto.casttype) =
+    //      "github.com/tendermint/tendermint/crypto/multisig.PubKeyMultisigThreshold"];
```

Comment:
Question: What is holding back this variant? It seems that there is a `PubKeyMultiSigThreshold` message.

Reply:
running into quite a few cyclical imports here. Not sure how to tackle this type. If you have time in the coming days could we set up a call?

Reply:
After trying multiple strategies, the one that I have come to that is using the customtype extension which means we will need these methods on the multisig: https://github.com/gogo/protobuf/blob/master/custom_types.md#custom-type-method-signatures

Reply:
there was also discussion around the cosmos-sdk moving keys to the app, the ones that they use. If this is the case then multisig would be removed.